person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..9999)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
```

Comment:
- [ ] Good job so far but I would like to recommend you to declared your attributes before your method The attr method creates an instance variable and a getter method for each attribute name passed as argument . Check the example below and please fix it in order to make your app work as expected :+1: 

```
class Person 


  attr_accessor :name, :age

  attr_reader :id

  def initialize(age, name = 'Unkown', parent_permission: true)
    @id = Random.rand(1..1000)
    @name = name
    @age = age
    @parent_permission = parent_permission
  end

--------
The rest of your code 
--------
```